src/v0/destinations/fb_custom_audience/recordTransform.js

```diff
@@ -0,0 +1,277 @@
+/* eslint-disable no-const-assign */
+const lodash = require('lodash');
+const get = require('get-value');
+const {
+  InstrumentationError,
+  ConfigurationError,
+  getErrorRespEvents,
+} = require('@rudderstack/integrations-lib');
+const { schemaFields } = require('./config');
+const { MappedToDestinationKey } = require('../../../constants');
+const stats = require('../../../util/stats');
+const {
+  getDestinationExternalIDInfoForRetl,
+  isDefinedAndNotNullAndNotEmpty,
+  checkSubsetOfArray,
+  returnArrayOfSubarrays,
+  getSuccessRespEvents,
+  generateErrorObject,
+} = require('../../util');
+const {
+  ensureApplicableFormat,
+  getUpdatedDataElement,
+  getSchemaForEventMappedToDest,
+  batchingWithPayloadSize,
+  responseBuilderSimple,
+  getDataSource,
+} = require('./util');
+
+function getErrorMetaData(inputs, acceptedOperations) {
+  const metadata = [];
+  // eslint-disable-next-line no-restricted-syntax
+  for (const key in inputs) {
+    if (!acceptedOperations.includes(key)) {
+      inputs[key].forEach((input) => {
+        metadata.push(input.metadata);
+      });
+    }
+  }
+  return metadata;
+}
+
+const processRecordEventArray = (
+  recordChunksArray,
+  userSchema,
+  isHashRequired,
+  disableFormat,
+  paramsPayload,
+  prepareParams,
+  destination,
+  operation,
+  operationAudienceId,
+) => {
+  const toSendEvents = [];
+  const metadata = [];
+  recordChunksArray.forEach((recordArray) => {
+    const data = [];
+    recordArray.forEach((input) => {
+      const { fields } = input.message;
+      let dataElement = [];
+      let nullUserData = true;
+
+      userSchema.forEach((eachProperty) => {
+        const userProperty = fields[eachProperty];
+        let updatedProperty = userProperty;
+
+        if (isHashRequired && !disableFormat) {
+          updatedProperty = ensureApplicableFormat(eachProperty, userProperty);
+        }
+
+        dataElement = getUpdatedDataElement(
+          dataElement,
+          isHashRequired,
+          eachProperty,
+          updatedProperty,
+        );
+
+        if (dataElement[dataElement.length - 1]) {
+          nullUserData = false;
+        }
+      });
+
+      if (nullUserData) {
+        stats.increment('fb_custom_audience_event_having_all_null_field_values_for_a_user', {
+          destinationId: destination.ID,
+          nullFields: userSchema,
+        });
+      }
+      data.push(dataElement);
+      metadata.push(input.metadata);
+    });
+
+    const prepareFinalPayload = lodash.cloneDeep(paramsPayload);
+    prepareFinalPayload.schema = userSchema;
+    prepareFinalPayload.data = data;
+    const payloadBatches = batchingWithPayloadSize(prepareFinalPayload);
```

Comment:
Is payload size limitation in batch a new requirement from FB ?

Reply:
this was present in audience flow as well